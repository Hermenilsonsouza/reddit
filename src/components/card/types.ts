export type CardProps = {
  title: string;
  authorFullname: string;
  ups: number;
  numComments: number;
  url?: string;
  created: number;
};
