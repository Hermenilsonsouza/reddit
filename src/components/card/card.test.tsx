import React from 'react';
import {render} from '@testing-library/react-native';
import Card from './card';
import {CardProps} from './types';
import moment from 'moment';

const makeSut = (otherProps?: Partial<CardProps>) => {
  const props: CardProps = {
    authorFullname: 't2_63val2uo',
    title: 'any_title',
    onPress: jest.fn(),
    link: 'any_link',
    ups: 44889,
    numComments: 6,
    created: 1676472000.0,
    url: 'https://b.thumbs.redditmedia.com/Kcl7QHZgBChhHrGhue6f6VNXMFLtqiKpBYGODPFJnFU.jpg',
    ...otherProps,
  };
  const component = render(<Card {...props} />);
  return {component, props};
};

describe('Card', () => {
  it('there is snapshot when have a image', () => {
    const {component} = makeSut();
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('there is not snapshot when have a image', () => {
    const {component} = makeSut({url: undefined});
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should there is a image', () => {
    const {component} = makeSut();
    const img = component.getByTestId('image');

    expect(img).toBeTruthy();
  });

  it('should there is not a image', () => {
    const {component} = makeSut({url: undefined});
    const img = component.queryByTestId('image');

    expect(img).toBeFalsy();
  });

  it('should there is a all texts', () => {
    const {component} = makeSut();
    const title = component.getByText('any_title');
    const author = component.getByText('t2_63val2uo');

    expect(title).toBeTruthy();
    expect(author).toBeTruthy();
  });

  it('should there is a formated date', () => {
    const {component, props} = makeSut();
    const date = component.queryByTestId('date');

    expect(date).toBeTruthy();

    const formated = moment(props.created).calendar();
    expect(date?.children[0]).toBe(formated);
  });
});
