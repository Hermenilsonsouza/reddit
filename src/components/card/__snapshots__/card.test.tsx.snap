// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card there is not snapshot when have a image 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "flexDirection": "row",
      "height": 110,
      "marginBottom": 8,
      "marginLeft": 8,
      "marginRight": 8,
      "marginTop": 8,
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      [
        {
          "height": "100%",
          "width": "25%",
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "space-between",
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#111",
            "fontSize": 12,
            "marginRight": 8,
            "textAlign": "right",
          },
        ]
      }
      testID="date"
    >
      01/20/1970
    </Text>
    <Text
      style={
        [
          {
            "color": "#111",
            "fontSize": 18,
            "fontWeight": "bold",
          },
        ]
      }
    >
      any_title
    </Text>
    <View
      style={
        [
          {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "95%",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#111",
              "fontSize": 12,
            },
          ]
        }
      >
        t2_63val2uo
      </Text>
      <Text
        style={
          [
            {
              "color": "#111",
              "fontSize": 12,
            },
          ]
        }
      >
        Score: 
        44889
      </Text>
      <Text
        style={
          [
            {
              "color": "#111",
              "fontSize": 12,
            },
          ]
        }
      >
        6
         comments
      </Text>
    </View>
  </View>
</View>
`;

exports[`Card there is snapshot when have a image 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "flexDirection": "row",
      "height": 110,
      "marginBottom": 8,
      "marginLeft": 8,
      "marginRight": 8,
      "marginTop": 8,
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      [
        {
          "height": "100%",
          "width": "25%",
        },
      ]
    }
  >
    <Image
      source={
        {
          "uri": "https://b.thumbs.redditmedia.com/Kcl7QHZgBChhHrGhue6f6VNXMFLtqiKpBYGODPFJnFU.jpg",
        }
      }
      style={
        [
          {
            "height": "100%",
            "width": "100%",
          },
        ]
      }
      testID="image"
    />
  </View>
  <View
    style={
      [
        {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "space-between",
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#111",
            "fontSize": 12,
            "marginRight": 8,
            "textAlign": "right",
          },
        ]
      }
      testID="date"
    >
      01/20/1970
    </Text>
    <Text
      style={
        [
          {
            "color": "#111",
            "fontSize": 18,
            "fontWeight": "bold",
          },
        ]
      }
    >
      any_title
    </Text>
    <View
      style={
        [
          {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "95%",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#111",
              "fontSize": 12,
            },
          ]
        }
      >
        t2_63val2uo
      </Text>
      <Text
        style={
          [
            {
              "color": "#111",
              "fontSize": 12,
            },
          ]
        }
      >
        Score: 
        44889
      </Text>
      <Text
        style={
          [
            {
              "color": "#111",
              "fontSize": 12,
            },
          ]
        }
      >
        6
         comments
      </Text>
    </View>
  </View>
</View>
`;
